waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -386,8 +386,28 @@ when defined(rln) or (not defined(rln) and not defined(rlnzerokit)):
 
     return proofBytes
 
-  proc proofVerify*(rlnInstance: RLN[Bn256], data: openArray[byte],
-    proof: RateLimitProof): bool =
+  proc getMerkleRoot*(rlnInstance: RLN[Bn256]): MerkleNodeResult =
+    # read the Merkle Tree root after insertion
+    var
+      root {.noinit.}: Buffer = Buffer()
+      rootPtr = addr(root)
+      get_root_successful = get_root(rlnInstance, rootPtr)
+    if (not get_root_successful): return err("could not get the root")
+    if (not (root.len == 32)): return err("wrong output size")
+
+    var rootValue = cast[ptr MerkleNode] (root.`ptr`)[]
+    return ok(rootValue)
+
+  proc validateVerifiedProof*(rlnInstance: RLN[Bn256], proof: RateLimitProof): bool =
+    # Validate against the local merkle tree
+    let localTreeRoot = rlnInstance.getMerkleRoot().value()
```

Comment:
If you have an expectation that `getMerkleRoot()` could return an error result (and based on the above it seems like it could happen), you shouldn't access the `MerkleNodeResult` value directly (which would raise a fatal `Defect`). I assume an error Result would be another indication that `validateVerifiedProof` fails?

Reply:
Addressed in [a0b3761](https://github.com/status-im/nwaku/pull/1158/commits/a0b376199382ddf0174a2319187d5d5399ac4f1d)